katana/grib_crop_wgrib2.py

```diff
@@ -170,79 +183,85 @@ def create_new_grib(start_date, end_date, directory, out_dir,
-                            fp,
-                            out_dir,
-                            file_time,
-                            x1, y1,
-                            logger,
-                            buff=buff,
-                            zone_letter=zone_letter,
-                            zone_number=zone_number,
-                            nthreads_w=nthreads_w)
-
-                        # proceed and break when we get a good file
-                        if sgrib:
-                            # grab just the variables we need
-                            sgrib_variable_crop(tmp_grib, nthreads_w,
-                                                fp_out, logger)
-                            break
-
-                        # kill job if we didn't find a good file
-                        if fx_hr == 6:
-                            raise IOError('No good grib file for {}'.format(
-                                file_time.strftime('%Y-%m-%d %H')))
-
-                # track hours per day
-                counter += 1
-
-            else:
-                logger.warning(
-                    '{} is not in date range and \
-                        will not be included'.format(file_time))
-
-        num_list.append(counter)
-
-    return date_list, num_list
+    fx_start_hour = 0
+
+    logger.info('Creating new gribs for topo domain')
+
+    # create a datelist dict to hold the files
+    out_files = {}
+    for dt in date_list:
+        if dt.date() not in out_files.keys():
+            out_files[dt.date()] = 0
+
+    # option to cut down on already completed work
+    if make_new_gribs:
+
+        # loop through dates
+        for idt, dt in enumerate(date_list):
+            logger.info('Working on grib file for {}'.format(dt))
+
+            # try different forecast hours to get a working file
+            for fx_hr in range(fx_start_hour, fx_start_hour+8):
+                fp = hrrr_file_name_finder(directory, dt, fx_hr)
+
+                # convert grib to smaller gribs with only the needed
+                # variables for WindNinjaS
+                sgrib, tmp_grib, fp_out = grib_to_small_grib(
+                    fp,
+                    out_dir,
+                    dt,
+                    x1, y1,
+                    logger,
+                    buff=buff,
+                    zone_letter=zone_letter,
+                    zone_number=zone_number,
+                    nthreads_w=nthreads_w)
+
+                # proceed and break when we get a good file
+                if sgrib:
+                    # grab just the variables we need
+                    status = sgrib_variable_crop(tmp_grib, nthreads_w,
+                                                 fp_out, logger)
+                    if status:
+                        out_files[dt.date()] += 1
+                        break
+
+                # kill job if we didn't find a good file
+                if fx_hr == 6:
```

Comment:
Looks like this leaves an hour unchecked, since

fx_hr in range(fx_start_hour, fx_start_hour+8)

Reply:
Yup, this was how it was in the original so I wasn't really changing it here. #6 will rework this so we can specify what hour to start at instead of always starting at 0.